orocos_kdl/src/tree.cpp

```diff
@@ -156,7 +156,7 @@ bool Tree::getChain(const std::string& chain_root, const std::string& chain_tip,
     }
 
     // add the segments from the common frame to the tip frame
-    for (int s=parents_chain_tip.size()-1; s>-1; s--){
+    for (int s=static_cast<int>(parents_chain_tip.size())-1; s>-1; s--){
```

Comment:
Why not go for an `unsigned int` here?

Reply:
The condition for the loop is to check if `s>-1`, so it must be able to become negative.

Reply:
And if we check for `s>=0`¿

Reply:
If `s` is an `unsigned int` it cannot be negative, so `s--` will cause an integer overflow and this will become an infinite loop.

I've updated the code to use a reverse iterator, which I think makes it more readable ([3324420](https://github.com/orocos/orocos_kinematics_dynamics/pull/390/commits/3324420efff9b58b53d37680beb01a81e440d6f6)).